pkg/controllers/management/authprovisioningv2/role.go

```diff
@@ -153,6 +157,83 @@ func (h *handler) OnChange(key string, rt *v3.RoleTemplate) (*v3.RoleTemplate, e
 	return rt, nil
 }
 
+func (h *handler) hasClusterAnnotationsAndDeletionTimestamp(annotations map[string]string) (bool, error) {
```

Comment:
A general comment on the name of this function - this doesn't really describe what this function is doing. The current name of the function implies that you are validating that a selection of annotations is present in the map - namely a cluster-related annotation and a deletion timestamp.

In reality, this function is to validate (through annotations and some other calls) that you have a binding that is granting access to a deleted cluster.

My suggestion would be to rename this to `isAnnotationsForDeletingCluster` (or something similar). You can also leave a comment above the function indicating the purpose of the function and why we care about this.

Reply:
I struggled with the name, and expected someone bringing it up, I like your suggestion more than mine.